src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java

```diff
@@ -1356,4 +1390,41 @@ public Descriptor<OicServerConfiguration> getDefaultServerConfigurationType() {
             return Jenkins.get().getDescriptor(OicServerWellKnownConfiguration.class);
         }
     }
+
+    /**
+     * Obtain {@code uri} as a string if it is not {@code null} otherwise returns the {@code defaultUri}.
+     * @param uri the possibly null URI
+     * @param defaultUri the URI to use if {@code uri} is null.
+     */
+    private static String fixNullUri(URI uri, String defaultUri) {
+        if (uri != null) {
+            return uri.toASCIIString();
+        }
+        return defaultUri;
+    }
+
+    /**
+     * Obtain the first auth method that we support that is supported by the server.
+     */
+    private static TokenAuthMethod fixTokenAuthMethod(
+            List<ClientAuthenticationMethod> supportedProviderMethods, TokenAuthMethod defaultProviderMethod) {
+        if (supportedProviderMethods == null || supportedProviderMethods.isEmpty()) {
+            return defaultProviderMethod;
+        }
+        return supportedProviderMethods.stream()
+                .map(OicSecurityRealm::toSupportedAuthMode)
+                .filter(Objects::nonNull)
+                .findFirst()
+                .orElse(defaultProviderMethod);
+    }
+
+    private static TokenAuthMethod toSupportedAuthMode(ClientAuthenticationMethod auth) {
+        String value = auth.getValue();
+        for (TokenAuthMethod tam : TokenAuthMethod.values()) {
+            if (tam.toString().equals(value)) {
+                return tam;
+            }
+        }
+        return null;
+    }
```

Comment:
is this used anywhere?

Reply:
Seems only `toSupportedAuthMode`, the other 2 methods seem not used.
Anyway, these changes come from a push in the `pac4j` branch, so I would say maybe better check if they're used there.